frontend/common/src/components/context/NestedLanguageProvider.tsx

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import { IntlProvider, useIntl } from "react-intl";
+import { useSearchParams } from "react-router-dom";
+
+import defaultRichTextElements from "../../helpers/format";
+import { Messages } from "./LanguageProvider";
+
+/**
+ * Indigenous Language codes
+ *
+ * crg - Michif
+ * crk - Plains Cree
+ * ojw - Western Ojibway
+ * mic - Mikmaq
+ */
+export type SecondaryLocale = "crg" | "crk" | "ojw" | "mic";
+
+interface NestedLanguageProvider {
+  messages: Map<string, Messages>;
+  children: React.ReactNode;
+}
+
+/**
+ * Nested Language Provider
+ *
+ * Providers a secondary language within the
+ * existing Language Provider.
+ *
+ * Note: This must be used within a LanguageProvider
+ */
+const NestedLanguageProvider = ({
+  messages,
+  children,
+}: NestedLanguageProvider) => {
+  const { messages: fallbackMessages } = useIntl();
+  const [searchParams] = useSearchParams();
+  const locale = searchParams.get("locale") as SecondaryLocale;
```

Comment:
I don't love this cast.  It doesn't actually check if the provided string matches a locale, right?  It might be better to check and then force it to null if it doesn't fit.

Reply:
Oh, is this what you mean about the maps being safer?  In that case maybe get rid of the SecondaryLocale type since it's not really providing any value?  Or do you have plans for it down the road.  :smile:

Reply:
Yeah, so we can check to make sure it is a locale but with the use of maps, `messages` would return undefined and exit early if the key does not exists.

The plan is the use the type for the map to constrain that. I should have included that in the test instructions but I was more focused on getting a MVP up for testing 😅 

<img width="907" alt="Screenshot 2022-11-21 at 4 55 15 PM" src="https://user-images.githubusercontent.com/4127998/203165972-d830bb75-801a-4e4d-9c35-0557bfa3e25b.png">

Reply:
As an example of the early exit:

https://github.com/GCTC-NTGC/gc-digital-talent/blob/221af6de0dc20418118461a34d84c0d7c2c63dad/frontend/common/src/components/context/NestedLanguageProvider.tsx#L37-L44

If we pass something like "not-a-locale" in as a search param, `locale` would be "not-a-locale" and `localeMessages` would be `undefined` so line 42 should resolve to true and just render the children without the provider so nothing changes.

Though, we could write something like `isSecondaryLocale` and add that to the checks as well.

```tsx
export type SecondaryLocale = "crg" | "crk" | "ojw" | "mic";
const secondaryLocales = ["crg", "crk", "ojw", "mic"];

export function isSecondaryLocale(
  locale?: string | null,
): locale is SecondaryLocale {
  return locale ? secondaryLocales.includes(locale) : false;
}
```

Reply:
Yeah, I like the type guard idea or just leave locale as a string.  It makes me uncomfortable to do an explicit cast when we know it's not actually constrained to that type.

Reply:
Alright, updated with [add type guard instead of casting](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4821/commits/04e1b47b07c77fa2a7c9f7b359e8578dabb6e3d9).

I'm not a fan of maintaining a type and an array but I honestly couldn't figure out a better way to type guard a type alias like this 🤷‍♀️

Reply:
Okay, removed all the extra stuff in [remove secondary type and type guard](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4821/commits/637a28d9ec352df4b9cb9e7ae18ae80623cfc7da).

If this looks good to you, I think we can merge.

Reply:
This looks great!  Thanks for being patient with me.  :grin: